File: Day227_2025/Q227_GFG.java
import java.util.*;

class Q227_GFG {
    public ArrayList<int[]> insertInterval(int[][] intervals, int[] newInterval) {
        int start = newInterval[0], end = newInterval[1];
        ArrayList<int[]> ans = new ArrayList<>();
        int c = -1;
        int n = intervals.length;
        
        for (int i = 0; i < n; i++) {
            int a = intervals[i][0], b = intervals[i][1];
            if (b < start) {
                ans.add(new int[]{a, b});
            } else if (c == -1 && Math.max(a, start) <= Math.min(b, end)) {
                start = Math.min(start, a);
                end = Math.max(end, b);
            } else {
                if (c == -1) {
                    c = 0;
                    ans.add(new int[]{start, end});
                }
                ans.add(new int[]{a, b});
            }
        }
        if (c == -1) {
            ans.add(new int[]{start, end});
        }
        return ans;
    }
}

File: Day227_2025/Q227_GFG.py
class Solution:
    def insertInterval(self, intervals, newInterval):
        start, end = newInterval
        ans = []
        c = -1
        
        for a, b in intervals:
            if b < start:
                ans.append([a, b])
            elif c == -1 and max(a, start) <= min(b, end):
                start = min(start, a)
                end = max(end, b)
            else:
                if c == -1:
                    c = 0
                    ans.append([start, end])
                ans.append([a, b])
        
        if c == -1:
            ans.append([start, end])
        return ans
